iModelCore/GeomLibs/geom/test/PolyfaceTest/t_ClipPlaneSet.cpp

```diff
@@ -2361,3 +2361,193 @@ TEST(ClipPlaneSet, AlignmentByMatchedArrays)
+    bool ret = clipPlaneSet.ClipCurveVector(*curve, clippedRegion);
+    Check::Shift(30,0,0);
+    Check::True(ret);
+    for (CurveVectorPtr clippedCurve : clippedRegion)
+        Check::SaveTransformed(*clippedCurve);
+    Check::ClearGeometry("ClipPlaneSet.ClipNotOpenBoundary");
+    }
+
+/*---------------------------------------------------------------------------------**//**
+* @bsimethod
++---------------+---------------+---------------+---------------+---------------+------*/
+TEST(ClipPlaneSet,ClipComplexOpenBoundary)
+    {
+    // curve
+    CurveVectorPtr curve = CurveVector::Create(CurveVector::BOUNDARY_TYPE_Open);
+    DEllipse3d arc = DEllipse3d::FromPointsOnArc(
+        DPoint3d::From(0, -5, 0),
+        DPoint3d::From(8, 8, 0),
+        DPoint3d::From(0, 5, 0)
+        );
+    ICurvePrimitivePtr arcPrim = ICurvePrimitive::CreateArc(arc);
+    curve->push_back(arcPrim);
+
+    bvector<DPoint3d> lineString1 = {
+        DPoint3d::From(0, 5, 0),
+        DPoint3d::From(-1, 10, 0),
+        DPoint3d::From(-2, 5, 0),
+        DPoint3d::From(-3, 10, 0),
+        DPoint3d::From(-4, 5, 0),
+    };
+    ICurvePrimitivePtr lineString1Prim = ICurvePrimitive::CreateLineString(lineString1);
+    curve->push_back(lineString1Prim);
+
+    DSegment3d seg1 = DSegment3d::From(DPoint3d::From(-4, 5, 0), DPoint3d::From(-10, 0, 0));
+    ICurvePrimitivePtr seg1Prim = ICurvePrimitive::CreateLine(seg1);
+    curve->push_back(seg1Prim);
+    DSegment3d seg2 = DSegment3d::From(DPoint3d::From(-10, 0, 0), DPoint3d::From(-4, -5, 0));
+    ICurvePrimitivePtr seg2Prim = ICurvePrimitive::CreateLine(seg2);
+    curve->push_back(seg2Prim);
+
+    bvector<DPoint3d> lineString2 = {
+        DPoint3d::From(-4, -5, 0),
+        DPoint3d::From(-3, -10, 0),
+        DPoint3d::From(-2, -5, 0),
+        DPoint3d::From(-1, -10, 0),
+        DPoint3d::From(0, -5, 0),
+    };
+    ICurvePrimitivePtr lineString2Prim = ICurvePrimitive::CreateLineString(lineString2);
+    curve->push_back(lineString2Prim);
+    Check::SaveTransformed(curve);
+
+    // clipper
+    bvector<DPoint3d> poly2 {{-7,-7,0},{7,-7,0},{7,7,0},{-7,7,0},{-7,-7,0}};
+    Check::SaveTransformed(poly2);
+    ConvexClipPlaneSet convexClipPlaneSet;
+    convexClipPlaneSet.ReloadSweptConvexPolygon(poly2, DVec3d::From(0, 0, 1), 0);
+    ClipPlaneSet clipPlaneSet(convexClipPlaneSet);
+    // perform the clip
+    bvector<CurveVectorPtr> clippedRegion;
+    bool ret = clipPlaneSet.ClipCurveVector(*curve, clippedRegion);
+    Check::Shift(30,0,0);
+    Check::True(ret);
+    for (CurveVectorPtr clippedCurve : clippedRegion)
+        Check::SaveTransformed(*clippedCurve);
+    Check::ClearGeometry("ClipPlaneSet.ClipComplexOpenBoundary");
+    }
+
+/*---------------------------------------------------------------------------------**//**
+* @bsimethod
++---------------+---------------+---------------+---------------+---------------+------*/
+TEST(ClipPlaneSet,ClipNonPlanarOpenBoundary)

```

Comment:
Same with these last two tests.  If you combine the tests, you can compare their results and get an additional geometry check. Since inputs are the same except in z, the results should be the same after flattening the non-planar result. To flatten, you can use a projection to the xy-plane = a Transform with matrix: (row0 = 100, row1 = 010, row2 = 000), translation = 000, then call `clipped1[i]->TransformInPlace(flatten)`.